### PR TITLE
EDSC-3634: As a cmr-graphql user, I want to retrieve Data Quality Summaries

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Adding data-quality-summaries to be discoverable via graphql including all of it's fields. Also adding data-quality-summaries as discoverable via a direct association from collection concepts

### What is the Solution?
Adding type definition, resolver, datasource, and extending the collection resolver to retrieve DQS concepts from the association-details

Summarize what you changed.

Extends the collection queries to include finding dataQualitySummaries from single or multiple collections (The EDSC use-case). Also allows for querying for DQS records directly as well.

### What areas of the application does this impact?

Resolvers, handler, datasources, and minor addition to the util function to parse concepts 

# Testing

### Reproduction steps

Go to the graphql apollo studio

SIT

send this query into graphql apollo studio
```
{
  dataQualitySummaries {
    items {
      conceptId
      associationDetails
      name
      summary
      id 
    }
  }
}
```

### Attachments

No relevant attachments

# Checklist

- [X] I have added automated tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: src/resolvers/__tests__/collection.test.js [hunk 1]
```diff
@@ -1514,6 +1516,205 @@ describe('Collection', () => {
       })
     })
 
+    describe('data-quality-summaries', () => {
+      describe('no associations are present in the metadata', () => {
+        test('doesn\'t query for or return variables', async () => {
```
Comment (macrouch) [inline src/resolvers/__tests__/collection.test.js:1521]: 'doesn\'t query for or return **variables**'

File: src/types/dataQualitySummary.graphql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "The list of DataQualitySummary search results."
+  items: [DataQualitySummary]
+}
+
+input DataQualitySummariesInput {
+  "The unique concept-id assigned to the DQS."
+  conceptId: [String]
+  "Cursor that points to the a specific position in a list of requested records."
+  cursor: String
+  "The number of variables requested by the user."
```
Comment (macrouch) [inline src/types/dataQualitySummary.graphql:30]: The number of **variables** requested by the user.

File: src/types/dataQualitySummary.graphql [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  limit: Int
```
Comment (macrouch) [inline src/types/dataQualitySummary.graphql:31]: If we are providing a `limit` parameter, shouldn't we also provide `offset`?

Comment (eudoroolivares2016) [reply src/types/dataQualitySummary.graphql:31]: Yes... we should! A few other concepts I'll add those too

File: src/types/dataQualitySummary.graphql [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  "The native id to set on the DQS."  
+  nativeId: String
+  "contains the human readable summary of the quality of the data set."
+  summary: String!
+}
+
+type DataQualitySummaryList {
+  "The number of hits for a given search."
+  count: Int
+  "Cursor that points to the a specific position in a list of requested records."
```
Comment (zimzoom) [inline src/types/dataQualitySummary.graphql:19]: the / a specific position

File: src/types/dataQualitySummary.graphql [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  cursor: String
+  "The list of DataQualitySummary search results."
+  items: [DataQualitySummary]
+}
+
+input DataQualitySummariesInput {
+  "The unique concept-id assigned to the DQS."
+  conceptId: [String]
+  "Cursor that points to the a specific position in a list of requested records."
```
Comment (zimzoom) [inline src/types/dataQualitySummary.graphql:28]: the / a specific position

**Review by zimzoom — APPROVED** [review]

File: src/types/dataQualitySummary.graphql (hunk 5 above)
Comment (eudoroolivares2016) [reply src/types/dataQualitySummary.graphql:28]: Great catch I'll adjust this and the other concepts 😄

File: src/cmr/concepts/dataQualitySummary.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    const { items } = data
+
+    return items
+  }
+
+  /**
+   * Query the CMR UMM API endpoint to retrieve requested data
+   * @param {Object} searchParams Parameters provided by the query
+   * @param {Array} requestedKeys Keys requested by the query
```
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply src/cmr/concepts/dataQualitySummary.js:31]: Also a great catch I'm gonna change this for all the concepts to make things consistent

**Review by macrouch — APPROVED** [review]

**Pull request merged**